### PR TITLE
Ignore /32 routes when programming VXLAN blackhole routes.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

/32 blackholes are inherently racey vs the CNI plugin adding a workload.
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
Fix race where felix programs a /32 blackhole route for a borrowed IP address concurrently with CNI plugin programming a "proper" route.
```

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]

**Review by petercork — APPROVED** [review]

File: dataplane/linux/vxlan_mgr.go [hunk 1]
```diff
@@ -216,6 +217,29 @@ func (m *vxlanManager) OnUpdate(protoBufMsg interface{}) {
...
+		return false
+	}
+	// Ignore /32 routes in any case for two reasons:
+	// * If we have a /32 block then our blackhole route would stop the CNI plugin from programming its /32 for a
+	//   newly added workload.
+	// * If this isn't a /32 block then it must be a borrowed /32 from another block.  In that case, we know we're
+	//   racing with CNI, adding a new workload.  We've received the borrowed IP but not the workload endpoint yet.
+	if strings.HasSuffix(msg.Dst, "/32") {
+		return false
+	}
```
Comment (petercork) [inline dataplane/linux/vxlan_mgr.go:240]: might need a `return true` here

Comment (fasaxc) [reply dataplane/linux/vxlan_mgr.go:240]: Haha, yep.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**